src/main/java/io/cdap/plugin/salesforce/plugin/source/batch/SalesforceBulkRecordReader.java

```diff
@@ -214,7 +246,8 @@ private String[] waitForBatchResults(BulkConnection bulkConnection)
       }
 
       if (info.getState() == BatchStateEnum.Completed) {
-        return bulkConnection.getQueryResultList(jobId, batchId).getResult();
+        return Failsafe.with(SalesforceSplitUtil.getRetryPolicy(initialRetryDuration, maxRetryDuration, maxRetryCount))
+          .get(() -> getQueryResultList(bulkConnection));
```

Comment:
I am wondering if we would need to unwrap those unchecked exceptions to prevent unneeded layers / cluttered message, see: https://github.com/data-integrations/bigquery-delta-plugins/pull/40/files#diff-427b4fbd30b2c80fb381e0e8fc3a6e2424ec9f0c94e8ed0881b0bc49402fc385R268-R279 for reference.

Reply:
Needs to be changed at all the three places where it is used or here only?

Reply:
The comment applies to all similar places.

Reply:
Added